cmd/lncli/cmd_macaroon.go

```diff
@@ -470,6 +411,86 @@ func printMacaroon(ctx *cli.Context) error {
 	return nil
 }
 
+// applyMacaroonConstraints parses and applies all currently supported macaroon
+// condition flags from the command line to the given macaroon and returns a new
+// macaroon instance.
+func applyMacaroonConstraints(ctx *cli.Context,
+	mac *macaroon.Macaroon) (*macaroon.Macaroon, error) {
+
+	macConstraints := make([]macaroons.Constraint, 0)
+
+	if ctx.IsSet(macTimeoutFlag.Name) {
+		timeout := ctx.Int64(macTimeoutFlag.Name)
+		if timeout <= 0 {
+			return nil, fmt.Errorf("timeout must be greater than 0")
+		}
+		macConstraints = append(
+			macConstraints, macaroons.TimeoutConstraint(timeout),
+		)
+	}
+
+	if ctx.IsSet(macIpAddressFlag.Name) {
+		ipAddress := net.ParseIP(ctx.String(macIpAddressFlag.Name))
+		if ipAddress == nil {
+			return nil, fmt.Errorf("unable to parse ip_address: %s",
+				ctx.String("ip_address"))
+		}
+
+		macConstraints = append(
+			macConstraints,
+			macaroons.IPLockConstraint(ipAddress.String()),
+		)
+	}
+
+	if ctx.IsSet(macCustomCaveatNameFlag.Name) {
+		customCaveatName := ctx.String(macCustomCaveatNameFlag.Name)
+		if containsWhiteSpace(customCaveatName) {
+			return nil, fmt.Errorf("unexpected white space found " +
+				"in custom caveat name")
+		}
+		if customCaveatName == "" {
+			return nil, fmt.Errorf("invalid custom caveat name")
+		}
+
+		var customCaveatCond string
+		if ctx.IsSet(macCustomCaveatConditionFlag.Name) {
+			customCaveatCond = ctx.String(
+				macCustomCaveatConditionFlag.Name,
+			)
+			if containsWhiteSpace(customCaveatCond) {
+				return nil, fmt.Errorf("unexpected white " +
+					"space found in custom caveat " +
+					"condition")
+			}
+			if customCaveatCond == "" {
+				return nil, fmt.Errorf("invalid custom " +
+					"caveat condition")
+			}
+			if customCaveatCond != "" && customCaveatName == "" {
```

Comment:
Ah, you're correct of course.
This also reminded me to update the usage notation to `[--custom_caveat_name= [--custom_caveat_condition=]]` since we only check the custom caveat condition if a custom caveat name is set in the first place.